ecs-cli/modules/cli/compose/entity/service/service.go

```diff
@@ -35,21 +36,25 @@ import (
 // Service type is placeholder for a single task definition and its cache
 // and it performs operations on ECS Service level
 type Service struct {
-	taskDef          *ecs.TaskDefinition
-	cache            cache.Cache
-	ecsContext       *context.ECSContext
-	timeSleeper      *utils.TimeSleeper
-	deploymentConfig *ecs.DeploymentConfiguration
-	loadBalancer     *ecs.LoadBalancer
-	role             string
-	healthCheckGP    *int64
+	taskDef           *ecs.TaskDefinition
+	cache             cache.Cache
+	ecsContext        *context.ECSContext
+	timeSleeper       *utils.TimeSleeper
+	deploymentConfig  *ecs.DeploymentConfiguration
+	loadBalancer      *ecs.LoadBalancer
+	role              string
+	healthCheckGP     *int64
+	serviceRegistries []*ecs.ServiceRegistry
 }
 
 const (
 	ecsActiveResourceCode  = "ACTIVE"
 	ecsMissingResourceCode = "MISSING"
 )
 
+// make servicediscovery.Create easily mockable in tests
+var servicediscoveryCreate = servicediscovery.Create
```

Comment:
This seems like an okay time to use a var as a mock, but see below comment in test.

Reply:
Yeah I think this is an example of a time when it makes things simpler